### PR TITLE
Added a new readme that much better describes the project

### LINKED ISSUE #12: README update
update README to include information on what is generated, including a diagram of the folder structure

### PR DESCRIPTION
**Pull request by CrimsonNynja**
This rewrites most of the readme file to properly describe this project.
included are better steps to run the generated code, a description of what is generated, and a short tutorial on how to run all the graphql querys that are generated.

This also resolves #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,13 +21,91 @@ Then generate your new project:
 yo node-graphql
 ```
 
+## Generated code
+
+If all default options are entered, the code generated should include a few runnable tests, as well as a runnable server.
+
+### Tests
+
+The generated tests are written using `Jest`, and provide an in memory database using `mongodb-memory-server`, as well as means to test any graphql resolvers and schema validations using `easygraphql-tester`.
```
Comment (dozyhermit) [inline README.md]: "using `Jest`, and provide an in memory database using..." => "using `Jest` and provide an in memory database using..."

File: README.md [hunk 2]
```diff
@@ -21,13 +21,91 @@ Then generate your new project:
...
+
+The server can be run with nodemon by:
+
+```bash
+npm run dev
+```
+
+### Database
+
+By default (and currently only choice), the chosen database is noSQL, which uses the `mongoose` package. While the tests can run fine without any setup here, to properly run the server, an instance of `mongoDB` should be created for the system to connect to. Please check the `.env` file and place your database credentials there.
```
Comment (dozyhermit) [inline README.md]: "By default (and currently only choice), the chosen database is noSQL, which uses the `mongoose` package" => "By default the chosen database is noSQL, which uses the `mongoose` package (keep in mind noSQL is currently the only option you can use)"

"to properly run the server, an instance of `mongoDB` should be created for the system to connect to." => "to properly run the server an instance of `mongoDB` should be created for the system to connect to."

File: README.md [hunk 3]
```diff
@@ -21,13 +21,91 @@ Then generate your new project:
...
+
+### What the code can do
+
+Out of the box, if you have already set up a database, run the server and navigate to `localhost:<PORT>/graphql`
+
+Out of the box, the system provides a method of creating a user, logging in, and checking the logged in user, all of this, using JWT for auth. If you navigate to the `user.graphql` file, you can see the provided mutations. Feel free to extend this here for your own system. But for now, if we want to create a user, we can simply enter the following in graphql playground
```
Comment (dozyhermit) [inline README.md]: "Out of the box, the system provides a method of creating a user, logging in, and checking the logged in user, all of this, using JWT for auth." => "Out of the box, the system provides methods for creating a user, logging in and checking the logged in user (all methods using JWT's for auth)."

File: README.md [hunk 4]
```diff
@@ -21,13 +21,91 @@ Then generate your new project:
...
+
+```gql
+mutation {
+  signup(username: "user", email: "test@email.com", password: "password")
+}
+```
+
+Which will create a new user and return us a token.
+
+From here, we can either use that token, or test the login, which we can do as follows
```
Comment (dozyhermit) [inline README.md]: "From here, we can either use that token, or test the login, which we can do as follows" => "From here we can either use that token or test the login, which we can do like this"

File: README.md [hunk 5]
```diff
@@ -21,13 +21,91 @@ Then generate your new project:
...
+
+```gql
+mutation {
+  login(email: "test@email.com", password: "password")
+}
+```
+
+Which also returns us a token. Now lets try check who is currently logged in with our token.
```
Comment (dozyhermit) [inline README.md]: "Which also returns us a token. Now lets try check who is currently logged in with our token." => "The above also returns us a `<type of token (refresh. etc)>` token. So now, let's verify and check who is currently logged in with our token."

**Review by dozyhermit — CHANGES_REQUESTED** [review]
makes sense! just some grammar things to make it easier to read.

File: README.md (hunk 2 above)
Comment (dozyhermit) [reply README.md]: also if people have no idea how to do the above, link to mongoDB install guide?

Comment (CrimsonNynja) [reply README.md]: I might put up another task to link all the packages to their git or npm registries and include this in that

**Pull request merged**